### DISCUSSION TITLE
What would a good integration with Model Context Protocol (MCP) look like

### DISCUSSION THREAD

**Discussion by systemshift** [wavetermdev/waveterm#1451] [Ideas]
One thing I immediately thought of when I used Wave for the first time was that this would make the perfect use case to integrate with LLM models. A model having a general view of varous terminals and blocks can be a powerful human-machine hybrid use case, such as using a terminal on localhost, ssh terminal, code editor, and a web browser block all as one context.
Antrhopic recently made MCP open source and I am considering to test it with Wave. https://modelcontextprotocol.io/introduction

So asking what would a good demo look like

**Comment by Mazyod** [comment]
A few simple MCPs for git, filesystem access could unleash the power of Wave AI pane. A demo would be to ask Claude to review the git changes, commit, perform certain updates, run tests, etc.

If you provide just shell access as an MCP server, it could manage installations, convert files, organization files/folders, etc.

I'm honestly very impressed with the amazing features in Wave out of the box, but the MCP is a missing piece that stands out.

**Comment by CrazyAce25** [comment]
Is there any word on this?  Right now the only MCP compatible terminals are Gemini CLI, which is tied to a google account and Gemini or Warp, which isnt open.  I'd like to integrate Wave into Theia AI IDE via MCP.

**Comment by dcieslak19973** [comment]
I just found waveterm as I was investigating Warp; Warp now supports MCP, would be nice to see the same functionality here

**Comment by dmgm** [comment]
i've been using gemini-cli to assist me with compiling software like hyprland. 

waveterm looks to be the scaffolding that will let me do the same thing with ollama. 

very exciting. 
plus bonus points for choosing.
